### ISSUE TITLE
Way to compress message selectively

### ISSUE THREAD

**Issue by qyangdu** [boostorg/beast#1757]
When use permessage deflate, compresion works better for large message than small message. Could Beast support a message size threshold parameter to enable compression for large messages only?

**Comment by vinniefalco** [comment]
I agree, there should be a way to turn compression on or off for each message.

**Comment by jcmonnin** [comment]
I think the "on off for each message" that @vinniefalco describes covers that, but here is my use case:
Most the peers are in the same subnet over a gigabit connection, and compression slows down the transfer noticeably for them. There are a few remote hosts where I would like to enable compression. I would like to be able to turn compression on/off depending in the peer address. 
A message size threshold as initially suggested is not flexible enough.

**Comment by vinniefalco** [comment]
>  I would like to be able to turn compression on/off depending in the peer address.
You can do that now, fill out a `permessage_deflate` and call `stream::set_option` with it.

**Comment by jcmonnin** [comment]
Thanks. That looks straightforward. I could have figured that out by thinking before writing...

**Comment by klemens-morgenstern** [comment]
Is this still unresolved?

**Comment by vinniefalco** [comment]
I believe we did add the per-message compression flag, can we please check?

**Comment by qyangdu** [comment]
The current per-message compression flag is a stream-level setting.
For better performance, the feature requested is to ask to compress message selectively, e.g. to compress message with size > 500 bytes only.

**Comment by ashtum** [comment]
Addressed in https://github.com/boostorg/beast/commit/bd69638e9d45fccbfdbb5eb0fae8c6cd4140a196.
